lib/gst/player/gstplayer.c

```diff
@@ -1660,6 +2154,163 @@ gst_player_get_pipeline (GstPlayer * self)
   return val;
 }
 
+/**
+ * gst_player_get_media_info:
+ * @player: #GstPlayer instance
+ *
+ * Returns: (transfer full): The internal media info instance.
+ */
+GstPlayerMediaInfo*
+gst_player_get_media_info (GstPlayer *self)
+{
+  return self->priv->media_info;
```

Comment:
Need to return a new reference here, and also needs locking (also in the other code where you actually set the media info)